fameal/ape-tax
# Feature/addVault Script

## What it does ✨
Add a script to be able to safely add a vault to the json file.
This script works with a default interactive way and support multiple option (to be easy to integrate with another script).
It will prompt for the needed elements and edit the `vaults.json` file to append the new vault.

Some elements will be fetched from the desired desired blockchain and from coingecko (`WANT_SYMBOL`, `COINGECKO_SYMBOL`, `WANT_ADDR`) and a `--fast` option is available to set  `VAULT_ABI`, `VAULT_TYPE` and `VAULT_STATUS` to their default value.

## How to test ✅
To add a new vault to the list, you need to run the following command : `node scripts/newVault.js --fast`.  
This will prompt you with several information:  
- What's the name of your vault ? (*ex: Hardrock Farmer*)
- What's the logo for your vault ? (*ex: 🎸👨‍🌾*)
- Which chain ? (*oneOf: Mainnet (1), BSC (56), Polygon (137), Fantom Opera (250)*)
- What's the address of your vault ? (*ex: 0x33bd0f9618cf38fea8f7f01e1514ab63b9bde64b*)
- Who is the dev of this vault ? (*ex: emilianobonassi*)

<img width="482" alt="Capture d’écran 2021-08-04 à 00 26 37" src="https://user-images.githubusercontent.com/9974362/128094349-de173732-ec31-4314-9d34-b73221a9099f.png">

You can also use the non-interactive mode: 
```
node scripts/newVault.js --name="Hardrock Farmer" --logo=🌾🌾 --chain=1 --address=0xFD0877d9095789cAF24c98F7CCe092fa8E120775 --dev=emilianobonassi --abi=yVaultV2 --type=experimental --status=active
```

Then, a new vault should be available in the JSON file (if the name of the address is not already used)

## Notes ⚠️
- Depending on your terminal, all the emojies may not be supported
- ~We should add a `--coingeckoID` option to overwrite the coingecko id for non-supported tokens (ex: LP tokens). This is not currently supported with the non-interactive mode.~
